processor/src/test/java/org/mapstruct/ap/test/subclassmapping/SubclassMappingTest.java

```diff
@@ -79,6 +95,24 @@ void subclassMappingInheritsMapping() {
             .containsExactly( "BenZ" );
     }
 
+    @ProcessorTest
+    @WithClasses( {
+        HatchBack.class,
+        InverseOrderSubclassMapper.class
+    } )
+    void subclassMappingOverridesInverseInheritsMapping() {
+        VehicleCollection vehicles = new VehicleCollection();
+        Car car = new Car();
+        car.setVehicleManufacturingCompany( "BenZ" );
+        vehicles.getVehicles().add( car );
+
+        VehicleCollectionDto result = InverseOrderSubclassMapper.INSTANCE.map( vehicles );
```

Comment:
Don't we want to test the inverse mapping here and check the type of the generated objects?

Reply:
Indeed, fixed the tests.